packages/mobile/src/screens/notifications-screen/Reaction/ReactionList.tsx

```diff
@@ -120,7 +120,11 @@ export const ReactionList = (props: ReactionListProps) => {
 
   const handleMeasure: OnMeasure = useCallback((config) => {
     const { x, width, reactionType } = config
-    positions.current = { ...positions.current, [reactionType]: { x, width } }
+    // Sometimes this function is called before the View being measured is laid out,
+    // so check that x and width are valid before updating the positions.
+    if (x > 0 && width > 0) {
```

Comment:
I wonder if `useLayoutEffect` would be useful here? It's supposed to run synchrnously _after_ the DOM has been set up for the component. So it should help ensure the layout is valid? Not sure if it behaves the same in ReactNative, though.

Reply:
Yeah I had the same thought, tried it and behavior was the same as `useEffect` so I bailed on it. Tbh I don't fully understand why we're getting 0 values here and also sometimes when we call it during `onLayout` bc the component should be laid out afaik, maybe we could discuss sometime but for now this is a bit of a hacky solution.